norduser/utmp.go

```diff
@@ -75,25 +59,51 @@ import (
 	"github.com/NordSecurity/nordvpn-linux/internal"
 )
 
-func getActiveUsers() ([]string, error) {
-	var usersCArray **C.char
+type userData map[string]norduserState
+
+// getActiveUsers returns a map of [username]userType where type can be text or gui. If any of the given users login
+// processes will be detected to have a gui(done based on the environment), user type will be gui. Otherwise user type
+// will be text.
+func getActiveUsers() (userData, error) {
+	var usersCArray *C.user
 	size := C.get_utmp_user_processes(&usersCArray)
+	defer C.free(unsafe.Pointer(usersCArray))
 
 	if size == C.ERROR_REALLOC {
-		return []string{}, fmt.Errorf("failed to reallocate space for the users table")
+		return userData{}, fmt.Errorf("failed to reallocate space for the users table")
 	} else if size == C.ERROR_MALLOC_USERNAME {
-		return []string{}, fmt.Errorf("failed to allocate space for new user in the users table")
+		return userData{}, fmt.Errorf("failed to allocate space for new user in the users table")
 	}
 
 	log.Printf("%s %d active user processes found", internal.DebugPrefix, size)
 
-	users := []string{}
-	for i := 0; i < int(size); i++ {
-		usernameCStr := (**C.char)(unsafe.Pointer(uintptr(unsafe.Pointer(usersCArray)) + uintptr(i)*unsafe.Sizeof(*usersCArray)))
-		username := C.GoString(*usernameCStr)
-		users = append(users, username)
+	users := make(userData)
+	for index := 0; index < int(size); index++ {
+		userC := (*C.user)(unsafe.Pointer(uintptr(unsafe.Pointer(usersCArray)) +
+			uintptr(index)*unsafe.Sizeof(*usersCArray)))
+		usernameC := (*C.char)(unsafe.Pointer(&userC.username))
+		username := C.GoString(usernameC)
+
+		// userType was determined to be gui for any of the users login processes, so we skip this user.
+		if userType, ok := users[username]; ok && userType == loginGUI {
+			continue
+		}
+
+		loginPID := uint32(userC.login_pid)
+
+		desktopSession, err := findEnvVariableForPID(loginPID, "XDG_CURRENT_DESKTOP")
```

Comment:
I think we need to define a list of supported desktop env for the app documentation. Because not all have this defined.

Reply:
Added tech debt task